### DISCUSSION TITLE
Automatic detection of new devices on the LAN

### DISCUSSION THREAD

**Discussion by Electra99** [arendst/Tasmota#16658] [Q&A] [answered]
What protocols could I used for automatic detection and identify new Tasmota devices on my LAN network using DHCP and not static IP-addresses ?
Pros and Cons for these broadcast protocols ?
Which protocol is the best ?
Is there any  Layer 3 broadcast IP-protocol ?
What Tasmota program module must I add in my firmware ?

I´m using Tasmota for all my SONOFF and Shelly devices with MQTT in my home.
My laptop server use Windows 10, Mosquitto and NodeRed.

**Comment by barbudor** [comment, marked as answer]
Tasmota announces itself on mqtt
You could detect tasmota devices by subscribing on `tasmota/discovery/+/config` like the below
```
08:28:43.481 MQT: tasmota/discovery/3C71BF2AD017/config = {"ip":"192.168.168.199","dn":"dev-4119","fn":["dev-4119",null,null,null,null,null,null,null],"hn":"dev-4119","mac":"3C71BF2AD017","md":"Generic","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"12.1.1.2","t":"dev-4119","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[1,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0],"so":{"4":1,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[0,0,0,0],"sht":[[0,0,0],[0,0,0],[0,0,0],[0,0,0]],"ver":1} (retained)
```

This is available in standard tasmota but obviously to work seamlessly you need your own custom binary with hard-coded wifi credentials and mqtt settings.
